File: frontend/src/service/dto/BookmarkDto.js
export default class BookmarkDto {
    id;
    logo;
    name;
    uri;
    parentFolderId;
    order;
    hierarchyParentFolderIds;

    constructor(Bookmark) {
        this.id = Bookmark.id;
        this.logo = null;
        this.name = Bookmark.name;
        this.uri = Bookmark.uri;
        this.parentFolderId = Bookmark.parentFolderId;
        this.order = Bookmark.order;
        this.hierarchyParentFolderIds = [Bookmark.parentFolderId];
    }

    static isBookmark(bookmark) {
        return bookmark instanceof BookmarkDto;
    }

    isDescendant(bookmark) {
        return this.hierarchyParentFolderIds.includes(bookmark.id);
    }
}

File: backend/src/main/java/com/bintage/pagemap/storage/application/dto/BookmarkDto.java
package com.bintage.pagemap.storage.application.dto;

import com.bintage.pagemap.storage.domain.model.bookmark.Bookmark;

import java.time.Instant;

public record BookmarkDto(long id,
                          long parentFolderId,
                          String name,
                          String uri,
                          String logo,
                          int order,
                          Instant createdAt) {

    public static BookmarkDto from(Bookmark bookmark) {
        return new BookmarkDto(bookmark.getId().value(),
                bookmark.getParentFolderId().value(),
                bookmark.getName(),
                bookmark.getUri().toString(),
                bookmark.getLogo().toString(),
                bookmark.getOrder(),
                bookmark.getCreatedAt());
    }
}
